jlm/llvm/ir/operators/lambda.hpp

```diff
@@ -21,46 +22,21 @@
 namespace jlm::llvm
 {
 
-namespace lambda
-{
-
 /** \brief Lambda operation
  *
  * A lambda operation determines a lambda's name and \ref rvsdg::FunctionType "function type".
  */
-class operation final : public rvsdg::StructuralOperation
+class LlvmLambdaOperation final : public rvsdg::LambdaOperation
```

Comment:
The namespace `jlm::llvm` clarifies it that this is the LLVM lambda, no? It could also just be named `LambdaOperation`.

Reply:
Naming it "LambdaNode" causes name resolution issues when both jlm::llvm and jlm::rvsdg are in scope. It also causes issues with moving code between modules (hence namespaces), as now suddenly instead of a compilation  error you get changed semantics. Lastly, it is also very ambiguous when reading the code unless you are using an IDE that constantly clarifies for every visible symbol where it resolves.

I was orginally inclined to name it more like "LlvmFunctionAttributes" or so (that's what it effectively is), rather going the direction of making it more ambiguous.